Mathlib/Topology/MetricSpace/Thickening.lean

```diff
@@ -631,4 +632,42 @@ theorem frontier_cthickening_disjoint (A : Set α) :
 
 end Cthickening
 
+theorem thickening_ball [PseudoMetricSpace α] (x : α) (ε δ : ℝ) :
+    thickening ε (ball x δ) ⊆ ball x (ε + δ) := by
+  rw [← thickening_singleton, ← thickening_singleton]
+  apply thickening_thickening_subset
+
 end Metric
+
+open Metric in
+theorem IsCompact.exists_thickening_image_subset
+    [PseudoEMetricSpace α] {β : Type*} [PseudoEMetricSpace β]
+    {f : α → β} {K : Set α} {U : Set β} (hK : IsCompact K) (ho : IsOpen U)
+    (hf : ∀ x : K, ContinuousAt f x) (hKU : MapsTo f K U) :
+    ∃ ε > 0, ∃ V ∈ 𝓝ˢ K, thickening ε (f '' V) ⊆ U := by
+  apply hK.induction_on (p := fun K ↦ ∃ ε > 0, ∃ V ∈ 𝓝ˢ K, thickening ε (f '' V) ⊆ U)
+  · use 1, by positivity, ∅, by simp, by simp
+  · exact fun s t hst ⟨ε, hε, V, hV, hthickening⟩ ↦ ⟨ε, hε, V, nhdsSet_mono hst hV, hthickening⟩
+  · rintro s t ⟨ε₁, hε₁, V₁, hV₁, hV₁thickening⟩ ⟨ε₂, hε₂, V₂, hV₂, hV₂thickening⟩
+    refine ⟨min ε₁ ε₂, by positivity, V₁ ∪ V₂, union_mem_nhdsSet hV₁ hV₂, ?_⟩
+    rw [image_union, thickening_union]
+    calc thickening (ε₁ ⊓ ε₂) (f '' V₁) ∪ thickening (ε₁ ⊓ ε₂) (f '' V₂)
+      _ ⊆ thickening ε₁ (f '' V₁) ∪ thickening ε₂ (f '' V₂) := by gcongr <;> norm_num
+      _ ⊆ U ∪ U := by gcongr
+      _ = U := union_eq_self_of_subset_left fun ⦃a⦄ a ↦ a
+  · intro x hx
+    have : {f x} ⊆ U := by rw [@singleton_subset_iff]; exact hKU hx
+    obtain ⟨δ, hδ, hthick⟩ := (isCompact_singleton (x := f x)).exists_thickening_subset_open ho this
+    let V := f ⁻¹' (thickening (δ / 2) { f x})
```

Comment:
I'm not sure if we prefer `{f x}` or `{ f x }`, but I'm sure it's not this. (Also below)

Reply:
Good point. I've checked, I think `{a}`for singleton sets is preferred.